ToyBox/classes/Infrastructure/CasterHelpers.cs

```diff
@@ -169,10 +170,34 @@ public static class CasterHelpers {
                 .Where(x => x.CharacterClass == spellbook.Blueprint.CharacterClass && (x.Archetype == null || unit.Progression.IsArchetype(x.Archetype))).Select(y => y.Spell)
                 .ToList();
 
-            return GetActualSpellsLearned(spellbook, level, spellsToIgnore);
+            Spellbook real = null;
+            if (unit.TryGetPartyMemberForLevelUpVersion(out var ch))
+            {
+                real = ch.Spellbooks.First(s => s.Blueprint == spellbook.Blueprint);
+            }
+
+            return GetActualSpellsLearned(spellbook, level, spellsToIgnore, real);
         }
 
-        public static int GetActualSpellsLearned(Spellbook spellbook, int level, List<BlueprintAbility> spellsToIgnore) {
+        public static int GetActualSpellsLearned(Spellbook spellbook, int level, List<BlueprintAbility> spellsToIgnore, Spellbook real = null) {
+            Func<AbilityData, bool> cond = x => true;
+            if (real != null)
+            {
+                var r = real.SureKnownSpells(level);
+                cond = x =>
```

Comment:
Please choose a meaningful identifier name that describes what the condition (so someone else understands your intention when they debug something in in later)

Reply:
Probably some comments explaining what this is doing would be helpful